nixos/modules/tasks/filesystems/zfs.nix

```diff
@@ -225,8 +225,18 @@ in
               esac
             done
             ''] ++ (map (pool: ''
-            echo "importing root ZFS pool \"${pool}\"..."
-            zpool import -d ${cfgZfs.devNodes} -N $ZFS_FORCE "${pool}"
+            echo -n "importing root ZFS pool \"${pool}\"..."
+            trial=0
+            until msg=$(zpool import -d ${cfgZfs.devNodes} -N $ZFS_FORCE "${pool}" 2>&1); do
+              sleep 0.25
+              echo -n .
+              trial=$(($trial + 1))
+              if [[ $trial -eq 60 ]]; then
+                echo $msg
+                break
+              fi
+            done
+            echo
```

Comment:
1. Yep. I doubt it'd be a problem, but if desired I can drop them into an array and print all of them on failure.
2. True. That's deliberate; it chops ~2 seconds off my boot sequence, and I don't *think* it'd break anything, but I'll admit I could be wrong.
3. I tested the overall code pretty thoroughly. What this line does is:
  - Redirect stderr to stdout (2>&1)
  - Put that output in $msg for later
  - And retry the import until it succeeds, of course.

Variable assignments have the same return code as the code they're wrapping, if any.